addon-test-support/backstop.js

```diff
@@ -156,13 +149,51 @@ function backstopHelper(name, options, res, err) {
   }, 0);
 }
 
+function createNameHash(assert, options) {
+  let name = options && options.name ? options.name : ""; //optional name to append to the assertion
+  let assertionName;
+  let testHash = {};
+
+  //namespace our steps to not conflict with qunit steps functionality
+  if (!assert.test.backstop) assert.test.backstop = {};
+  if (!assert.test.backstop.steps) assert.test.backstop.steps = [];
+
+  let steps = assert.test.backstop.steps;
+
+  if (name && name.length !== 0) {
+    name = ` | ${name}`;
+  }
+
+  // Generate base names to extend
+  if (
+    assert.test &&
+    assert.test.module &&
+    assert.test.module.name &&
+    assert.test.testName
+  ) {
+    testHash.testId = window._testRunTime;
+    testHash.scenarioId = assert.test.testId;
+    assertionName = `${assert.test.module.name} | ${assert.test.testName}`;
+  }
+
+  //Record our step
+  steps.push(steps.length);
```

Comment:
Just a question.  We're using an array here to essentially count.  Should we consider just an int here?  Instead of an array?   Not a big deal -- just wanted to know your thinking.  Thanks.

Reply:
good observation! yes and int would be better. It's residue from my first draft when I was using the built in steps method for tracking the state `assert.steps('my step name')`, which produced an array.

Reply:
changed to init.